src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.spec.ts

```diff
@@ -67,8 +91,96 @@ describe('PaypalCommerceButtonStrategy', () => {
+                        surname: 'Brown',
+                    },
+                    email_address: 'test@bigcommerce.com',
+                    payer_id: '123',
+                    address: {
+                        country_code: '123',
+                    },
+                },
+                purchase_units: [ {
+                    reference_id: '123',
+                    amount: {
+                        currency_code: 'USD',
+                        value: '100',
+                    },
+                    payee: {
+                        email_address: 'test@bigcommerce.com',
+                        merchant_id: '123',
+                    },
+                    shipping: {
+                        address: {
+                            address_line_1: '',
+                            address_area_1: '',
+                            address_area2: '',
+                            country_code: 'USD',
+                            postal_code: '968',
+                        },
+                        name: {
+                            full_name: 'ANDREW BROWN',
+                        },
+                    },
+                } ],
+        };
+
+        const onShippingChangeData = {
+            amount: {
+                breakdown: {
+                    item_total: {
+                        currency_code: 'USD',
+                        value: '100',
+                    },
+                shipping: {
+                    currency_code: 'USD',
+                    value: '100',
+                },
+                tax_total: {
+                    currency_code: 'USD',
+                    value: '100',
+                },
+                },
+                currency_code: 'USD',
+                value: '100',
+            },
+            orderID: '123',
+            payment_token: 'PAYMENT_TOKEN',
+            shipping_address: {
+                city: 'Los-Angeles',
+                postal_code: '08547',
+                country_code: 'US',
+                state: 'CA',
+            },
+            selected_shipping_option: {
+                id: '123',
+                amount: {
+                    currency_code: 'USD',
+                        value: '100',
+                },
+            },
+        };
+
         orderID = 'ORDER_ID';
         fundingSource = 'paypal';
+        actions = {
+            order: {
+                capture: jest.fn(() => new Promise(resolve => {
+                    return resolve({ ...payerDetails });
+                })),
+                authorize: jest.fn(() => new Promise(resolve => {
+                    return resolve({ ...payerDetails });
+                })),
+                patch: jest.fn(() => new Promise(resolve => {
+                    return resolve('TEST') ;
+                })),
+            },
+            resolve: jest.fn().mockReturnValue(Promise.resolve()),
+            reject: jest.fn().mockReturnValue(Promise.reject()),
+         };
```

Comment:
You don't need to use rest operator here, because `payerDetails` is already an object

```suggestion
        actions = {
            order: {
                capture: jest.fn(() => new Promise(resolve => {
                    return resolve(payerDetails);
                })),
                authorize: jest.fn(() => new Promise(resolve => {
                    return resolve(payerDetails);
                })),
                patch: jest.fn(() => new Promise(resolve => {
                    return resolve('TEST') ;
                })),
            },
            resolve: jest.fn().mockReturnValue(Promise.resolve()),
            reject: jest.fn().mockReturnValue(Promise.reject()),
         };
```

Reply:
fixed